From 83bb00bbc0f3ee5883db9b126e642881ecd9d0e5 Mon Sep 17 00:00:00 2001
From: Dorian Stoll <dorian.stoll@tmsp.io>
Date: Mon, 6 Apr 2020 13:57:25 +0200
Subject: [PATCH 02/10] ipts: Remove no_feedback quirk

Since the previous commit appears to fix the crashes on gen4 that
required no_feedback, quirking these devices is not neccessary anymore.

Signed-off-by: Dorian Stoll <dorian.stoll@tmsp.io>
---
 drivers/misc/ipts/companion/ipts-surface.c | 4 ++--
 drivers/misc/ipts/params.c                 | 4 ----
 drivers/misc/ipts/params.h                 | 1 -
 include/linux/ipts.h                       | 1 -
 4 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/misc/ipts/companion/ipts-surface.c b/drivers/misc/ipts/companion/ipts-surface.c
index 1a151538b898c..a86777c068e5a 100644
--- a/drivers/misc/ipts/companion/ipts-surface.c
+++ b/drivers/misc/ipts/companion/ipts-surface.c
@@ -34,13 +34,13 @@ struct ipts_surface_data {
 // Surface Book 1 / Surface Studio
 static const struct ipts_surface_data ipts_surface_mshw0076 = {
 	.hid = "MSHW0076",
-	.quirks = IPTS_QUIRK_NO_FEEDBACK,
+	.quirks = IPTS_QUIRK_NONE,
 };
 
 // Surface Pro 4
 static const struct ipts_surface_data ipts_surface_mshw0078 = {
 	.hid = "MSHW0078",
-	.quirks = IPTS_QUIRK_NO_FEEDBACK,
+	.quirks = IPTS_QUIRK_NONE,
 };
 
 // Surface Laptop 1 / 2
diff --git a/drivers/misc/ipts/params.c b/drivers/misc/ipts/params.c
index 93b19cbf47869..3ea76ca8342a9 100644
--- a/drivers/misc/ipts/params.c
+++ b/drivers/misc/ipts/params.c
@@ -18,7 +18,6 @@ struct ipts_params ipts_modparams = {
 	.ignore_fw_fallback = false,
 	.ignore_config_fallback = false,
 	.ignore_companion = false,
-	.no_feedback = -1,
 
 	.debug = false,
 	.debug_thread = false,
@@ -33,9 +32,6 @@ IPTS_PARAM(ignore_config_fallback, bool, 0400,
 IPTS_PARAM(ignore_companion, bool, 0400,
 	"Don't use a companion driver to load firmware. (default: false)"
 );
-IPTS_PARAM(no_feedback, int, 0644,
-	"Disable sending feedback to ME (can prevent crashes on Skylake). (-1=auto [default], 0=false, 1=true)"
-);
 
 IPTS_PARAM(debug, bool, 0400,
 	"Enable IPTS debugging output. (default: false)"
diff --git a/drivers/misc/ipts/params.h b/drivers/misc/ipts/params.h
index 4d9d2bca5edeb..c20546bacb086 100644
--- a/drivers/misc/ipts/params.h
+++ b/drivers/misc/ipts/params.h
@@ -15,7 +15,6 @@ struct ipts_params {
 	bool ignore_fw_fallback;
 	bool ignore_config_fallback;
 	bool ignore_companion;
-	int no_feedback;
 
 	bool debug;
 	bool debug_thread;
diff --git a/include/linux/ipts.h b/include/linux/ipts.h
index bfa8e1375926d..f229a34368516 100644
--- a/include/linux/ipts.h
+++ b/include/linux/ipts.h
@@ -15,6 +15,5 @@
 #define MAX_IOCL_FILE_PATH_LEN 256
 
 #define IPTS_QUIRK_NONE        0
-#define IPTS_QUIRK_NO_FEEDBACK BIT(0)
 
 #endif // IPTS_H
-- 
2.26.2

